### DISCUSSION TITLE
Hubbard correction implemented?

### DISCUSSION THREAD

**Discussion by TheoBardu** [phoebe-team/phoebe#206] [Q&A] [answered]
Hi! 
I would like to ask you a question.
I have tried to use the Hubbard correction on Phoebe-quantum espresso (i.e on the custom version of QE) but it seems to have no effect. Is it available?
Thank you so much, 
Theo

**Comment by jcoulter12** [comment]
Hi Theo, 

I do think this should be possible because of the way Phoebe handles using the el-ph matrix elements from QE. 
When you say it has no effect, what do you mean more specifically? No effect on the el-ph transport calculation? 

Best, 
Jenny

**Reply by TheoBardu** [reply]
I mean that for e.g, when performing scf calculation with the Hubbard parameters inserted, the output file of the scf does not contain the correction. There is no indication of the usage of the Hubbard correction, while in the original qe, it’s written the values of the energies of correction in the output. 
I don’t know if I’m wrong in doing something, I don’t have already checked if there is no effect in the el-ph transport calculation.
I just wanted to know if I were available the Hubbard correction in the scf calculation.
I apologise for the question.
Best regards, 
Theo

**Reply by jcoulter12** [reply]
Hi Theo, 

I haven't tested this using it in a while, let me ask a couple extra questions. I agree it's good to check things at this stage before moving on. 

1) Is the version of QE you're comparing to 7.0? Unfortunately, I haven't yet had the time to update the patched version of QE to the latest version yet, though to my knowledge nothing serious related to ph or elph caluclations has happened since v7. Perhaps what's printed to the output file may have changed, though. 
2) Does everything else remain exactly the same, like the total energy etc doesn't change when the U value is turned on vs. off in the Phoebe patched version? 

The only thing different about the patched version of Phoebe is that it writes the gauge information of the wavefunction to file. This creates some large files initially, but helps things to run faster at later stages of the calculation and allows us to natively use things like +U or different kinds of pseudopotentials. 
Therefore, I think the patch shouldn't change anything in the code related to the use of U and hopefully things should still work. 

In the worst case I can take a look at your input files quickly, and see if there's a reason U cannot be used that I'm not thinking of right away. 

No need to apologize for asking questions, either. 
Best, 
Jenny

**Reply by TheoBardu** [reply, marked as answer]
Thank you so much! 
Well, I have no results probably because I have used the syntax for QE7.1 that has the card HUBBARD at the end of the file, without remembering that you have modified the 7.0 version 😄 (if I remember). 
So probably I should use the previous syntax.
I was confused about that because no error message occurred. 
So, thank you so much!
Best, 
Theo

**Reply by jcoulter12** [reply]
Hi Theo, 

Thanks for letting me know about the syntax change, indeed, updating to the current QE version is on my todo list, just not the highest priority item yet. Glad things are (hopefully) working for you. 

If you need any other assistance with this calculation or see anything unexpected, let me know  -- as I mentioned, things should work with U but it's not heavily tested, so I'm curious as to how things turn out. 

Best, 
Jenny

**Reply by akshayphy00** [reply]
<img width="1038" height="402" alt="Screenshot 2026-09-03 111457" src="https://github.com/user-attachments/assets/9490b576-1076-4450-9c79-7b6c491de21c" />
I tried to run the calculation using the patched QE, but it shows that Hubbard‑U is not supported. Even I have also tried to run the phonon calculation by original QE(but 'scf' & 'nscf' are done by patched QE), still it shows the same. Is this feature not implemented? Could you please provide some insight on how I can proceed with my calculation using the Hubbard‑U correction?
